lib/read.js

```diff
@@ -258,12 +258,22 @@ class ReadSql extends AdapterRead {
                 _.each(points, function(pt) {
                     _.extend(pt, additionalFields);
                 });
-                return this.formatPoints(points);
+                return points;
             });
         }, { concurrency: BATCHING_CONCURRENCY })
         .then((res) => {
+            var points = _.chain(res)
+            .filter('length')
+            .sortBy(function(pts) {
+                return JSON.stringify(pts[0][timeField]);
```

Comment:
Ok, can you `flatten` before you `sortBy` and get rid of the `filter` and `[0]`?